pkg/dao/task_info.go

```diff
@@ -42,9 +42,8 @@ func (t taskInfoDaoImpl) Fetch(orgID string, id string) (api.TaskInfoResponse, e
 
 	result := t.db.Table(taskInfo.TableName()+" AS t ").
 		Select(JoinSelectQuery).
-		Joins("LEFT JOIN repositories r on t.repository_uuid = r.uuid").
-		Joins("LEFT JOIN repository_configurations rc on r.uuid = rc.repository_uuid").
-		Where("text(t.id) = ? AND t.org_id = ?", id, orgID).First(&taskInfo)
+		Joins("LEFT JOIN repository_configurations rc on t.repository_uuid = rc.repository_uuid").
+		Where("t.id = ? AND t.org_id = ?", UuidifyString(id), orgID).First(&taskInfo)
```

Comment:
So this makes sense but what would happen if the the id was "not-a-proper-uuid"? 
I take it the UUIDifyString fucntion would return a uuid2.Nil. 
Would we want that in our database?
Would it be prudent to do an error check prior to the sql call?

Reply:
this is just for querying, we'd never insert a uuid.nil value.

Reply:
we could add an error check for each query to validate that its a proper UUID and return an empty set, but thats a bit more obtuse i think.  The bigger issue is that its easy to write a query that doesn't handle this well.  Maybe the longer term solution is to accept UUIDs at the api level?   I'll check if thats even possible but thats probably a bigger longer term solution and this is a good short term performance fix